consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/ConsumerGatewayTest.java

```diff
@@ -0,0 +1,173 @@
+import static org.mockito.Mockito.*;
+
+/**
+ * Test class for the consumer gateway.
+ *
+ * @author Mark Dathorne
+ */
+@ExtendWith(MockitoExtension.class)
+class ConsumerGatewayTest extends SubmitQueryConsumerTest {
+
+    @Mock
+    IIndividualRequestConsumer individualRequestConsumer;
+
+    ConsumerGateway consumerGateway;
+
+    /**
+     * Method to do any pre-test set-up.
+     */
+    @BeforeEach
+    @Override
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        submitQueryConsumer = new SubQueryConsumer();
+        submitQueryConsumer.setTransformer(mockTransformer);
+        // submitQueryConsumer.setRethrowOnFailureToConnect (true);
+
+        submitQueryRequest = this.createSubmitQueryRequest();
+        submitQueryRequestType = this.createRequestType(submitQueryRequest);
+
+        consumerGateway = new ConsumerGateway();
+        consumerGateway.setIndividualRequestConsumer(individualRequestConsumer);
+        consumerGateway.setSubmitQueryConsumer(submitQueryConsumer);
+    }
+
+    /**
+     * Test to verify submit query consumer does throw expected exception.
+     */
+    @Test
+    @Override
+    void testSubmitQueryRequestTimeout() {
+        when(mockTransformer.transformDomainToJaxb(submitQueryRequest)).thenReturn(submitQueryRequestType);
+
+        final WebServiceException wsException = new WebServiceException();
+        wsException.initCause(new SocketTimeoutException("Timed out waiting for response"));
+
+        when(mockClient.submitQuery(submitQueryRequestType)).thenThrow(wsException);
+
+        TimeoutException timeoutException = assertThrows(TimeoutException.class, () ->
+                this.consumerGateway.submitQuery(submitQueryRequest, CONNECTION_TIME_OUT, RECEIVE_TIME_OUT));
+
+        assertEquals(timeoutException.getErrorCode(), "TIMEOUT_ERROR");
+        assertEquals(timeoutException.getErrorDescription(), "Read time out error sending [null]");
+        assertEquals(timeoutException.getCause(), null);
+    }
+
+    /**
+     * Test to verify submit query consumer does throw expected exception.
+     *
+     * @throws SOAPException exception
+     */
+    @Test
+    @Override
+    void testSubmitQueryRequestSoapFault() throws SOAPException {
+        when(mockTransformer.transformDomainToJaxb(submitQueryRequest)).thenReturn(submitQueryRequestType);
+
+        final WebServiceException wsException = new WebServiceException();
+        soapFault.setFaultCode("REQ_FAULT");
+        soapFault.setFaultString("Invalid request");
+
+        wsException.initCause(new SOAPFaultException(soapFault));
+
+        when(mockClient.submitQuery(submitQueryRequestType)).thenThrow(wsException);
+
+        SoapFaultException soapFaultException = assertThrows(SoapFaultException.class, () -> {
+            this.consumerGateway.submitQuery(submitQueryRequest, CONNECTION_TIME_OUT, RECEIVE_TIME_OUT);
+            assertEquals(this.soapFault.getFaultCode(),"REQ_FAULT");
```

Comment:
Although it technically doesn't matter, the assertEquals() parameters are the wrong way around.  The first is the expected value and the second is the actual value.  In this case "REQ_FAULT" is the expected value and this.soapFault.getFaultCode() is the actual value.  The same issue appears to be true of the other instances of assertEquals() in this class.

Reply:
This "issue" as a result of copy/paste of existing test class. Corrected now.